java_tests/src/test/java/org/lsst/sal/SAL_TestTest.java

```diff
@@ -0,0 +1,427 @@
+package org.lsst.sal;
+
+import Test.*;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class SAL_TestTest extends BaseTestCase {
+
+    @Test
+    public void testGetCurrentTime() {
+        salUtils su = new salUtils();
+        // Run the test twice in the unlikely event that the first run
+        // occurs at a leap second transition.
+        int leapSeconds = 0;
+        double measuredSeconds = 0.0;
+        for (int i = 0; i < 2; i++) {
+            leapSeconds = su.getLeapSeconds();
+            measuredSeconds = remote.getCurrentTime() - (System.currentTimeMillis() / 1000.0);
+            if (Math.abs(leapSeconds - measuredSeconds) - 1 < 0.1) {
+                // We are probably at a leap second transition;
+                // sleep and try again.
+                sleep(2);
+                continue;
+            }
+            break;
+        }
+        Assert.assertEquals(leapSeconds, measuredSeconds, 0.001);
+    }
+
+    @Test
+    @Ignore("No astropy equivalent for java.")
+    public void testGetLeapSeconds() {
+    }
+
+    @Test
+    public void testGetVersions() {
+        final String salVersion = remote.getSALVersion();
+        Assert.assertNotNull(salVersion);
+        Assert.assertNotSame("", salVersion);
+        final String xmlVersion = remote.getXMLVersion();
+        Assert.assertNotNull(xmlVersion);
+        Assert.assertNotSame("", xmlVersion);
+    }
+
+    /**
+     * Write several logevent messages and make sure gettting the oldest returns the data in the expected order.
+     *
+     * @throws Exception In case of a problem related to reflection.
+     */
+    @Test
+    @Ignore("Fails due to JNI limitations if ran with all other unit tests.")
```

Comment:
"ran" -> "run"

This is getting really scary. This is a very important test for serious problems we have had. Please try to find some way to kill the JVM between test invocations.

Perhaps using a different LSST_DDS_DOMAIN_PREFIX for each test invocation will fix this particular issue?

Reply:
See my previous remark about forking the JVM when running the unit tests. This issue will be addressed in DM-32914.